tests/lib/journalctl.sh

```diff
@@ -35,14 +35,20 @@ check_journalctl_ready(){
 check_journalctl_log(){
     expression=$1
     shift
-    for _ in $(seq 10); do
-        log=$(get_journalctl_log "$@")
-        if echo "$log" | grep -q -E "$expression"; then
+    # Initial checks dont print the log in the debug output
+    for iter in $(seq 9); do
+        if get_journalctl_log "$@" | grep -q -E "$expression"; then
```

Comment:
Silly question, so `echo "$log" | grep -q ....` makes the output appear in the log, but `get_journalctl_log "$@" | grep -q ...` doesn't?

Reply:
This makes the log appears in the debug output.
    log=$(get_journalctl_log "$@")
So the idea is to make 
    get_journalctl_log "$@" | grep -q -E "$expression" 
that pipes the log to the grep command instead

Reply:
I think Pawel's question is valid: both commands pipe to `grep -q` so why one logs and the other does not?
It seems to suggest `get_journalctl_log` is broken and that it writes to stderr.

A quick look at `get_journalctl_log` as well as `get_journalctl_log_from_cursor` doesn't show anything immediately broken. I will see what's going on here first.

Reply:
I had a look at this and proposed a quick sanity check PR: https://github.com/snapcore/snapd/pull/7225 -- let's see if that passes. It removes the whole loop entirely and simplifies surrounding code.